alf/drivers/off_policy_driver.py

```diff
@@ -232,6 +241,15 @@ def train(self,
                 batch = tf.nest.map_structure(
                     lambda x: x[b:tf.minimum(batch_size, b + mini_batch_size)],
                     experience)
+                # Make the shape explicit. The shapes of tensors from the
+                # previous line depend on tensor `b`, which is replaced with
+                # None by tf. This makes some operations depending on the shape
+                # of tensor fail. (Currently, it's alf.common.tensor_extend)
+                # TODO: Find a way to work around with shapes containing None
+                # at common.tensor_extend()
+                batch = tf.nest.map_structure(
+                    lambda x: tf.reshape(x, [mini_batch_size] + list(x.shape)[
+                        1:]), batch)
```

Comment:
Note: in another approved PR this has been changed to a Python loop. So make sure to make them consistent.